src/util/sharing_map.h

```diff
@@ -915,6 +954,18 @@ ::replace(const key_type &k, valueU &&m)
   lp->set_value(std::forward<valueU>(m));
 }
 
+SHARING_MAPT(void)
+::update(const key_type &k, std::function<void(mapped_type &)> mutator)
+{
+  innert *cp = get_container_node(k);
+  SM_ASSERT(cp != nullptr);
+
+  leaft *lp = cp->find_leaf(k);
+  PRECONDITION(lp != nullptr); // key must exist in map
+
+  lp->mutate_value(mutator);
```

Comment:
Can we add a debugging check here that tests if the mutated element is different from the old element? (if `fail_if_equal` is set, similar to the invariant in `replace()`)